### ISSUE TITLE
Memory failure in spite of available memory in slab.info()

### ISSUE THREAD

**Issue by alyapunov** [tarantool/tarantool#952]
Tarantool failed to complete an operation due to memory error:
```
box.space.requests.index.primary:delete{1276001}
---
- error: Failed to allocate 16384 bytes in mempool for new slab
...
```
Though box.slab.info() shows that more than 60Mb memory is available
```
 arena_used_ratio: 98.6%
 items_used_ratio: 97.1%
 arena_used: 4692856928
 arena_size: 4759819160
```

**Comment by kostja** [comment]
Alex, the problem of this bug is that index arena and tuple arenas are separate, and don't share slab caches. Thus, when you free data slabs, they can't be reused by index. The patch which implements a fix is on branch snapshot-crash. As you can see from the name, it crashes snapshot.test. Please find & fix the crash, and push the patch.

Thanks!

**Comment by alyapunov** [comment]
Index arena and tuple arena merged, #996 pending
